### PR TITLE
Add tests for wrangler tail

### PR DESCRIPTION
**Pull request by caass**
- ensure the correct API calls are made
- ensure that filters are sent
- ensure that the _correct_ filters are sent
- ensure that JSON gets spat out into the terminal

putting a changeset as well since some code changed in tail.tsx -- since there were errors caught by the tests!

### CODE REVIEW

File: packages/wrangler/src/__tests__/tail.test.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+      await expect(server.nextMessageJson()).resolves.toHaveProperty(
+        "filters",
+        [{ query: search }]
+      );
+    } finally {
+      server.close();
+    }
+  });
+
+  it("sends everything but the kitchen sink", async () => {
```
Comment (JacobMGEvans) [inline packages/wrangler/src/__tests__/tail.test.ts:217]: great test name 😆

**Review by nataliescottdavidson — COMMENTED** [review]
Great work @caass ! Good test structure and names. Big shoutout to all wrangler2 folks @caass @threepointone @JacobMGEvans @petebacondarwin and others for the AWESOME DevExp! This was my first time cloning the repo and running tests. Everything is so smooth and well documented. Good stuff!

**Review by petebacondarwin — APPROVED** [review]
LGTM!
Nice use of helpers and clean clean tests!

File: packages/wrangler/src/__tests__/tail.test.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    } finally {
+      server.close();
+      expect(tailDeletionRequests.count).toStrictEqual(1);
+    }
+  });
+
+  /* filtering */
+
+  it("sends sampling rate filters", async () => {
+    const { server } = setupWebsocketMocks();
```
Comment (threepointone) [inline packages/wrangler/src/__tests__/tail.test.ts:67]: every test has a setupWebSocketMocks() call and a finally clause that closes it. Maybe it should go to into a beforeEach/afterEach? that way you can avoid the try/finally as well.

**Review by threepointone — APPROVED** [review]
Glad to see that adding tests caught bugs, @petebacondarwin will be very happy to be proven right (as he always is)

Great work. Tentatively approving, but dropped some changes for before you land it.

**Pull request merged**